legend-engine-xts-connection/legend-engine-xt-connection-factory/src/main/java/org/finos/legend/connection/ConnectionExtension.java

```diff
@@ -28,4 +29,9 @@ default List<AuthenticationMechanismType> getExtraAuthenticationMechanismTypes()
     {
         return Collections.emptyList();
     }
+
+    default Map<String, DatabaseType> getExtraDatabaseTypeMapping()
```

Comment:
This is leaking relational concepts into the core.  This is an area we need to address long term, hence not sure we want to add new functionality that assume relational is part of core.  

For example - we should avoid a single enum for all db types since that conflicts with the modularization of the system.  

Anyway this can be part of the relational extension?

Reply:
@rafaelbey this is actually `DatabaseType` here is from `org.finos.legend.connection`, not the one from `storeRelational`, and it's designed with extensibility in mind: if you look at `RelationalDatabaseType`, that's an enum that extends this interface, hence, demonstrating we can further modularize this if we want into `SnowflakeRelationalDBType`, `BigQueryRelationalDBType`, etc.

I think perhaps I run out of names, but `database` to me applies to more than just relational DB, so when I name it `DatabaseType`, I envision we can do `DatabaseType.Mongo`, `DatabaseType.ElasticSearch` (also see `DatabaseSupport`, that's generic for all kinds of stores)

Thanks for reviewing :), but please check with @epsstan and @pierredebelen when/if we want to merge this.